node/toolrunner.ts

```diff
@@ -800,6 +800,15 @@ export class ToolRunner extends events.EventEmitter {
 
         let cp = child.spawn(this._getSpawnFileName(), this._getSpawnArgs(optionsNonNull), this._getSpawnOptions(options));
 
+        // it is possible for the child process to end its last line without a new line.
+        // because stdout is buffered, this causes the last line to not get sent to the parent
+        // stream. Adding this event forces a flush before the child streams are closed.
+        cp.stdout.on('finish', () => {
+            if (!optionsNonNull.silent) {
+                optionsNonNull.outStream.write(os.EOL);
```

Comment:
Hm, the only thing that I could think of that this might break is something like Jest Snapshots (or other similar snapshot based tools). I _think_ most of those tools are resilient to stuff like that, and it seems like a pretty unlikely use case anyways.

IMO as long as we can confirm that this fixes the original issue we should ship it

Reply:
Worst case I see is we get an extra new line in the logs (which actually maybe is the right thing anyways?)

Reply:
Yes, I confirmed it fixed the issue in my dev deployment. I could not get a unit test to fail as the symptom is with the stream between the agent and the task.